From a2d273c591adc32c7dc834bf8fef651affe172d0 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Wed, 4 Jan 2023 13:53:22 +0100
Subject: [PATCH 329/529] ARM: dts: stm32: add adc support on stm32mp135f-dk

Configure ADC support on stm32mp135f-dk. ADC can be used for
USB Type-C CC1 & CC2 pins wired to in6 & in12.

U-Boot enables ADC1 to support USB power measurement and ADC
calibration on STM32MP135x Disco boards.
When leaving U-boot the ADC does not return to power down state
to keep ADC linear calibration available for kernel.
Enable ADC1 by default on STM32MP135x-DK boards to align kernel
DT with Uboot. This avoids to shutdown the ADC VDDA, while the
ADC is not in power down.

Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/283315
Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Change-Id: I154f09b5218a4ef3e2b7db6aac9b9d997d7b0a73
---
 arch/arm/boot/dts/stm32mp135f-dk.dts | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

--- a/arch/arm/boot/dts/stm32mp135f-dk.dts
+++ b/arch/arm/boot/dts/stm32mp135f-dk.dts
@@ -128,6 +128,31 @@
 	};
 };
 
+&adc_1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&adc1_usb_cc_pins_a>;
+	vdda-supply = <&scmi_vdd_adc>;
+	vref-supply = <&scmi_vdd_adc>;
+	status = "okay";
+	adc1: adc@0 {
+		status = "okay";
+		/*
+		 * Type-C USB_PWR_CC1 & USB_PWR_CC2 on in6 & in12.
+		 * Use at least 5 * RC time, e.g. 5 * (Rp + Rd) * C:
+		 * 5 * (5.1 + 47kOhms) * 5pF => 1.3us.
+		 * Use arbitrary margin here (e.g. 5us).
+		 */
+		channel@6 {
+			reg = <6>;
+			st,min-sample-time-ns = <5000>;
+		};
+		channel@12 {
+			reg = <12>;
+			st,min-sample-time-ns = <5000>;
+		};
+	};
+};
+
 &dts {
 	status = "okay";
 };
